conan/packager.py

```diff
@@ -84,285 +107,96 @@ def __init__(self, args=None, username=None, channel=None, runner=None,
-                                self.add({"arch": arch,
-                                          "build_type": build_type,
-                                          "compiler": "gcc",
-                                          "compiler.version": gcc_version,
-                                          "compiler.libcxx": "libstdc++"},
-                                         {shared_option_name: shared})
-                                if float(gcc_version) > 5:
-                                    self.add({"arch": arch,
-                                              "build_type": build_type,
-                                              "compiler": "gcc",
-                                              "compiler.version": gcc_version,
-                                              "compiler.libcxx": "libstdc++11"},
-                                             {shared_option_name: shared})
-                            else:
-                                self.add({"arch": arch,
-                                          "build_type": build_type,
-                                          "compiler": "gcc",
-                                          "compiler.version": gcc_version},
-                                         {shared_option_name: shared})
-                else:
-                    for build_type in ["Debug", "Release"]:
-                        if not pure_c:
-                            self.add({"arch": arch,
-                                      "build_type": build_type,
-                                      "compiler": "gcc",
-                                      "compiler.version": gcc_version,
-                                      "compiler.libcxx": "libstdc++"}, {})
-                            if float(gcc_version) > 5:
-                                self.add({"arch": arch,
-                                          "build_type": build_type,
-                                          "compiler": "gcc",
-                                          "compiler.version": gcc_version,
-                                          "compiler.libcxx": "libstdc++11"}, {})
-                        else:
-                            self.add({"arch": arch,
-                                      "build_type": build_type,
-                                      "compiler": "gcc",
-                                      "compiler.version": gcc_version}, {})
-
-    def add(self, settings=None, options=None):
+        builds = []
+        if self._platform_info.system() == "Windows":
+            if self.mingw_configurations:
+                builds = get_mingw_builds(self.mingw_configurations, self.mingw_installer_reference, self.archs)
+            builds.extend(get_visual_builds(self.visual_versions, self.archs, self.visual_runtimes,
+                                            shared_option_name, dll_with_static_runtime, self.vs10_x86_64_enabled))
+        elif self._platform_info.system() == "Linux" or self.use_docker is True:
```

Comment:
Actually, it will ignore the docker option, only available for linux.
Running docker with OSX is possible, even in Windows, but it's not tested nor requested yet.